### PR TITLE
Updated claim ticket payment logic

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

File: launchpad-guaranteed-tickets/src/lib.rs [hunk 1]
```diff
@@ -212,7 +212,41 @@ pub trait LaunchpadGuaranteedTickets:
...
+                .direct(&owner, &ticket_price.token_id, 0, &claimable_ticket_payment);
+        }
+
+        let total_launchpad_tokens_deposited = self.total_launchpad_tokens_deposited().get();
+        let amount_per_ticket = self.launchpad_tokens_per_winning_ticket().get();
+        let total_nr_winning_tickets = claimable_ticket_payment / ticket_price.amount;
+
+        let total_launchpad_tokens_won = total_nr_winning_tickets * amount_per_ticket;
+
+        if total_launchpad_tokens_won > total_launchpad_tokens_deposited {
```
Comment (dorin-iancu) [inline launchpad-guaranteed-tickets/src/lib.rs:235]:
```suggestion
        if total_launchpad_tokens_won >= total_launchpad_tokens_deposited {
```

File: launchpad-guaranteed-tickets/src/lib.rs [hunk 2]
```diff
@@ -212,7 +212,41 @@ pub trait LaunchpadGuaranteedTickets:
...
+        }
+
+        let launchpad_token_id = self.launchpad_token_id().get();
+        let launchpad_tokens_balance = self.blockchain().get_esdt_balance(
+            &self.blockchain().get_sc_address(),
+            &launchpad_token_id,
+            0,
+        );
+        let extra_launchpad_tokens = total_launchpad_tokens_deposited - total_launchpad_tokens_won;
+        if extra_launchpad_tokens > 0 && extra_launchpad_tokens >= launchpad_tokens_balance {
```
Comment (dorin-iancu) [inline launchpad-guaranteed-tickets/src/lib.rs:246]:
```suggestion
        if extra_launchpad_tokens >= launchpad_tokens_balance {
```
I'm still not entirely sure you even need the second condition.

Also, fix tests.

File: launchpad-guaranteed-tickets/src/lib.rs (hunk 1 above)
Comment (psorinionut) [reply launchpad-guaranteed-tickets/src/lib.rs]: Updated.

File: launchpad-guaranteed-tickets/src/lib.rs (hunk 2 above)
Comment (psorinionut) [reply launchpad-guaranteed-tickets/src/lib.rs]: Updated.

**Review by dorin-iancu — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]

**Pull request merged**